### DISCUSSION TITLE
Using prezto-contrib w/ prezto

### DISCUSSION THREAD

**Discussion by blogabe** [mattmc3/antidote#170] [General]
Is there preferred way to import prezto-contrib? I can source it my zsh_plugins.txt file which will install is under `$ANTIDOTE_HOME/belak/prezo-contrib` but contrib instructions prefer to have it install at `$ANTIDOTE_HOME/sorin-ionescu/prezto/contrib`?

I like the first, default, option of having contrib install under $ANTIDOTE_HOME, but antidote is looking for a file it cannot find: `$ANTIDOTE_HOME/belak/prezto-contrib/contrib-prompt/contrib-prompt.plugin.zsh`

`zstyle ':prezto:load' pmodule-dirs $ANTIDOTE_HOME/belak/prezto-contrib` is set and here's how I'm calling it in zsh_plugins:
```
belak/prezto-contrib path:contrib-prompt/functions kind:autoload
belak/prezto-contrib path:contrib-prompt
```

**Comment by mattmc3** [comment]
### TLDR; Use this in your `.zsh_plugins.txt`

```zsh
belak/prezto-contrib path:contrib-prompt/functions kind:fpath
```

### Long answer:

As you probably know by now, antidote does not have any special handler for Prezto, Oh My Zsh, etc. Instead, it provides annotations to let you load plugins and frameworks however you choose. Thus, there's also no special logic in antidote to handle Prezto Contrib repos. Unless you choose to specifically load `sorin-ionescu/prezto` (meaning not really using antidote to manage your prezto plugins), directories like `$ANTIDOTE_HOME/sorin-ionescu/prezto/contrib` and zstyles like `zstyle ':prezto:load' pmodule-dirs $ANTIDOTE_HOME/belak/prezto-contrib` have no meaning to antidote are not used/necessary. Loading `sorin-ionescu/prezto` with antidote is not recommended - if you do this, you might as well use Prezto the conventional way.

> I like the first, default, option of having contrib install under $ANTIDOTE_HOME

Good, because that's the recommended way with antidote.

> but antidote is looking for a file it cannot find: $ANTIDOTE_HOME/belak/prezto-contrib/contrib-prompt/contrib-prompt.plugin.zsh

[If you look at how Kaleb has his contrib-prompt repo set up](https://github.com/belak/prezto-contrib/tree/a05508a716cad6e45e90cb8b0f73c811cbd4438a/contrib-prompt), there is no init.zsh file or contrib-prompt.plugin.zsh for that subplugin. That's because prompt plugins that use Zsh's `promptinit` system are a bit weird. Running code for a prompt plugin isn't strictly necessary because `promptinit` uses whatever functions are in `fpath` to determine what prompts are available. Meaning, you should not `autoload` of declare prompt functions - it's fpath only. This isn't just true for Kaleb's prezto-contrib - it's true for all prompt plugins that rely on `promptinit`:

```zsh
# .zsh_plugins.txt

# add prompt plugins to fpath before initializing prompt system
sindresorhus/pure kind:fpath
romkatv/powerlevel10k kind:fpath
belak/prezto-contrib path:contrib-prompt/functions kind:fpath

# now initialize prompt system
sorin-ionescu/prezto path:modules/prompt
```

So, TLDR: use this line: `belak/prezto-contrib path:contrib-prompt/functions kind:fpath`.

**Reply by blogabe** [reply]
Amazing. Thank you!
